sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/PeriodicSequence.java

```diff
@@ -223,11 +233,17 @@ public ProcessContinuation processElement(
           estimator.setWatermark(output);
           nextOutput = nextOutput + interval;
         }
+        if (!srcElement.catchUpToNow) {
+          break;
+        }
       }
 
       ProcessContinuation continuation = ProcessContinuation.stop();
       if (claimSuccess) {
-        Duration offset = new Duration(Instant.now(), Instant.ofEpochMilli(nextOutput));
+        Duration offset =
+            srcElement.catchUpToNow
+                ? new Duration(Instant.now(), Instant.ofEpochMilli(nextOutput))
+                : new Duration(interval);
```

Comment:
If catchUpToNow=False, then the emitted interval can be minutes old than now, is that true?

I think the real problem for current PeriodicImpulse is that it does not support a start time to be the time pipeline starts being processed. It needs to specify a constant start time, but we cannot control when pipeline will actually start processing

Is it possible to keep the emitted timestamp nearly now? Also the naming could be more formal (`catch up` sounds informal)

Reply:
>If catchUpToNow=False, then the emitted interval can be minutes old than now, is that true? 

That's right, this would allow having a start time of `new Instant(0)` and still respecting the fire interval.

> Is it possible to keep the emitted timestamp nearly now?

I think if a user needs this, they can just not set the flag. The current behavior does just that.